# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: startupcache/StartupCacheParent.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 startupcache/StartupCacheParent.cpp
--- a/startupcache/StartupCacheParent.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,32 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "mozilla/scache/StartupCacheParent.h"
-
-#include "mozilla/scache/StartupCache.h"
-#include "mozilla/dom/ContentParent.h"
-
-namespace mozilla {
-namespace scache {
-
-IPCResult StartupCacheParent::Recv__delete__(nsTArray<EntryData>&& entries) {
-  if (entries.Length()) {
-    auto* cache = StartupCache::GetSingleton();
-    for (auto& entry : entries) {
-      auto buffer = MakeUnique<char[]>(entry.data().Length());
-      memcpy(buffer.get(), entry.data().Elements(), entry.data().Length());
-      cache->PutBuffer(entry.key().get(), std::move(buffer),
-                       entry.data().Length(), /* isFromChildProcess:*/ true);
-    }
-  }
-
-  return IPC_OK();
-}
-
-void StartupCacheParent::ActorDestroy(ActorDestroyReason aWhy) {}
-
-}  // namespace scache
-}  // namespace mozilla
\ No newline at end of file